collect.py

```diff
@@ -119,25 +150,24 @@ def do_something(driver, elem_attributes=None):
             if not child.is_displayed() or not child.is_enabled():
                 continue
 
-            elem = child
-            break
+            res = check_attributes(driver, elem_attributes, child)  # returns list of elements with attributes = child_attributes
+            if len(res) == 1:
+                elem = child
+                break
+            else:
+                # remove all elements with the same attributes
+                print('Found elements with same attributes')
+                # print(res)
+                print()
+                for element in res:
+                    children.remove(element)
```

Comment:
You should never remove elements from the list you are iterating on. You can iterate over a copy and remove elements from the original.
```
lst = [1, 2, 1, 1, 3, 1, 1]
for elem in lst:
    if elem == 1:
        lst.remove(elem)
print(lst)
```
This will result in output as [2,3,1,1] and not [2,3]